tools/buildsys/src/gomod.rs

```diff
@@ -0,0 +1,160 @@
+/*!
+Packages using the Go programming language may have upstream tar archives that
+include only the source code of the project, but not the source code of any
+dependencies.  The Go programming language promotes the use of "modules" for
+dependencies and projects adopting modules will provide go.mod and go.sum
+files.
+
+This module provides the ability to retrieve and validate the dependencies
+declared using Go modules given a tar archive containing a go.mod and go.sum.
+
+The location where dependencies are retrieved from are controlled by the
+standard environment variables employed by the Go tool: GOPROXY, GOSUMDB, and
+GOPRIVATE.
+
+ */
+
+pub(crate) mod error;
+use error::Result;
+
+use super::manifest;
+use duct::cmd;
+use snafu::{OptionExt, ResultExt};
+use std::env;
+use std::path::PathBuf;
+use std::process::Output;
+
+pub(crate) struct GoMod;
+
+impl GoMod {
+    pub(crate) fn vendor(
+        root_dir: &PathBuf,
+        package_dir: &PathBuf,
+        gomods: &[manifest::GoModule],
+    ) -> Result<Self> {
```

Comment:
Looks like you can just return nothing in the happy case:
```suggestion
    ) -> Result<()> {
```